src/openms/include/OpenMS/ANALYSIS/XLMS/OpenProXLUtils.h

```diff
@@ -0,0 +1,302 @@
+
+    bool operator<(const PeptideMass& other) const
+    {
+      return peptide_mass < other.peptide_mass;
+    }
+
+    bool operator<(const double other) const
+    {
+      return peptide_mass < other;
+    }
+  };
+
+  struct PreprocessedPairSpectra
+  {
+    // pre-initialize so we can simply std::swap the spectra (no synchronization in multi-threading context needed as we get no reallocation of the PeakMapreprocessed_pair_spectra.
+    PeakMap spectra_common_peaks; // merge spectrum of common peaks (present in both spectra)
+    PeakMap spectra_xlink_peaks; // Xlink peaks in the light spectrum (common peaks between spectra_light_different and spectra heavy_to_light)
+    PeakMap spectra_all_peaks;
+
+    PreprocessedPairSpectra(Size size)
+    {
+      for (Size i = 0; i != size; ++i)
+      {
+        spectra_common_peaks.addSpectrum(PeakSpectrum());
+        spectra_xlink_peaks.addSpectrum(PeakSpectrum());
+        spectra_all_peaks.addSpectrum(PeakSpectrum());
+      }
+    }
+  };
+
+    //bool operator< (const double other, const XLPrecursor& pre) {return other < pre.precursor_mass;}
+
+    static float preScore(Size matchedAlpha, Size ionsAlpha, Size matchedBeta, Size ionsBeta);
+    static float preScore(Size matchedAlpha, Size ionsAlpha);
+
+//    static double binomialCoefficient(int n, int k);
+    static double cumulativeBinomial(Size n, Size k, double p);
+
+    static std::vector<XLPrecursor> enumerateCrossLinksAndMasses_(const std::vector<OpenProXLUtils::PeptideMass>&  peptides, double cross_link_mass_light, const DoubleList& cross_link_mass_mono_link, const StringList& cross_link_residue1, const StringList& cross_link_residue2, std::vector< double >& spectrum_precursors, double precursor_mass_tolerance, bool precursor_mass_tolerance_unit_ppm);
+
+
+    static double match_odds_score(const PeakSpectrum& theoretical_spec,  const std::vector< std::pair< Size, Size > >& matched_spec, double fragment_mass_tolerance, bool fragment_mass_tolerance_unit_ppm, bool is_xlink_spectrum, Size n_charges = 1);
+
+//    template <typename SpectrumType1, typename SpectrumType2>
+//    static std::vector< double > xCorrelation(const SpectrumType1 & spec1, const SpectrumType2 & spec2, Int maxshift, double tolerance);
+
+    static double weighted_TIC_score(Size alpha_size, Size beta_size, double intsum_alpha, double intsum_beta, double intsum, double total_current, bool type_is_cross_link);
+
+    // Sum of matched ion intesity, for Intsum score and %TIC score
+//    static double matched_current_chain(const std::vector< std::pair< Size, Size > >& matched_spec_common, const std::vector< std::pair< Size, Size > >& matched_spec_xlinks, const PeakSpectrum& spectrum_common_peaks, const PeakSpectrum& spectrum_xlink_peaks);
+
+
+    static void writeXQuestXML(String out_file, String base_name, const std::vector< PeptideIdentification >& peptide_ids, const std::vector< std::vector< CrossLinkSpectrumMatch > >& all_top_csms, const PeakMap& spectra,
```

Comment:
input/output handling is an obvious case that should be separated from the actual data processing. Please move to its own file in `OpenMS/ANALYSIS/XLMS/` or even better in `OpenMS/FORMAT/`